### DISCUSSION TITLE
why define driver?

### DISCUSSION THREAD

**Discussion by langziyang** [liip/LiipImagineBundle#1607] [General]
why define driver? how to automatic detection and use the best one ?

**Comment by dbu** [comment]
being explicit prevents surprises when e.g. the local environment has a different library than the server.

the current configuration picks gd as default if none is specified: https://github.com/liip/LiipImagineBundle/blob/f3b67426800a6a86840de7447bb552c61a79f4b6/DependencyInjection/Configuration.php#L109

we could change this to use a beforeNormalization method to pick the option that is available (and error if none is available). see https://symfony.com/doc/current/components/config/definition.html#normalization

do you want to do a pull request for this change?
